src/components/embedWrapper.tsx

```diff
@@ -0,0 +1,243 @@
+	};
+
+	const elementProps = parseDataAttributesFromElement(container);
+
+	switch (elementProps['kind'] as EmbedKind) {
+		case EmbedKind.Spotify:
+			return resultMap3(
+				(src: string, width: number, height: number): Spotify => ({
+					kind: EmbedKind.Spotify,
+					src,
+					width,
+					height,
+					source: fromNullable(elementProps['source']),
+					sourceDomain: fromNullable(elementProps['sourceDomain']),
+					tracking: parseTrackingParam(elementProps['tracking']),
+				}),
+			)(requiredStringParam(elementProps, 'src'))(
+				requiredNumberParam(elementProps, 'width'),
+			)(requiredNumberParam(elementProps, 'height'));
+		case EmbedKind.YouTube:
+			return resultMap3(
+				(id: string, width: number, height: number): YouTube => ({
+					kind: EmbedKind.YouTube,
+					id,
+					width,
+					height,
+					source: fromNullable(elementProps['source']),
+					sourceDomain: fromNullable(elementProps['sourceDomain']),
+					tracking: parseTrackingParam(elementProps['tracking']),
+				}),
+			)(requiredStringParam(elementProps, 'id'))(
+				requiredNumberParam(elementProps, 'width'),
+			)(requiredNumberParam(elementProps, 'height'));
+		case EmbedKind.Generic: {
+			return resultMap2<string, number, Generic>(
+				(html: string, height: number): Generic => ({
+					kind: EmbedKind.Generic,
+					alt: fromNullable(elementProps['alt']),
+					html,
+					height,
+					mandatory: elementProps['mandatory'] === 'true',
+					source: fromNullable(elementProps['source']),
+					sourceDomain: fromNullable(elementProps['sourceDomain']),
+					tracking: parseTrackingParam(elementProps['tracking']),
+				}),
+			)(requiredStringParam(elementProps, 'html'))(
+				requiredNumberParam(elementProps, 'height'),
+			);
+		}
+	}
+};
+
+const createEmbedComponentFromProps = (
+	container: Element,
+): Result<string, ReactElement> => {
+	return andThen((embed: Embed) => {
+		return resultFromNullable(
+			`I can't construct a Component for embed of type ${embed.kind}`,
+		)(h(EmbedComponentInClickToView, { embed }));
+	})(divElementPropsToEmbed(container));
+};
+
+const EmbedComponentInClickToView: FC<Props> = ({ embed }: Props) => {
+	return h(ClickToView, {
+		source: embed.source,
+		sourceDomain: embed.sourceDomain,
+		children: h(EmbedComponent, { embed }),
+		role: none,
+		onAccept: none,
+	});
+};
+
+const withDatasetKeyFormat = (
+	dataSet: Record<string, string>,
+): Record<string, string> => {
+	return Object.keys(dataSet).reduce((accumulatedObject, currentKey) => {
+		const newKey =
+			'data-' +
+			currentKey.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
+		return { ...accumulatedObject, [newKey]: dataSet[currentKey] };
+	}, {});
+};
+
+const EmbedComponentWrapper: FC<Props> = ({ embed }: Props) => {
+	if (
+		embed.tracking === EmbedTracksType.TRACKS ||
+		embed.tracking === EmbedTracksType.UNKNOWN
+	) {
+		return h(
+			'div',
+			{
+				...withDatasetKeyFormat(embedToDivProps(embed)),
+				className: 'click-to-view-container',
```

Comment:
Companion suggestion to a comment I made above.

```suggestion
				className: 'js-click-to-view-container',
```